Vinyl/Turntable.swift

```diff
@@ -85,32 +86,30 @@ public final class Turntable: NSURLSession {
     }
 
     public override var delegate: NSURLSessionDelegate? {
-        return nil
+        return _delegate
     }
 }
 
 // MARK: - NSURLSession methods
 
 extension Turntable {
+
+    public override func dataTaskWithURL(url: NSURL) -> NSURLSessionDataTask {
+        let request = NSURLRequest(URL: url)
+        return dataTaskWithRequest(request)
+    }
     
```

Comment:
**_tl;dr_** it will not work if I make that change

You are probably right. I think if I was designing the API that's probably what I'd do too. Unfortunately this method is required by the [`NSURLSession`](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/) class reference, the base class `Turntable` inherits.

Changing it to

```swift
public override func dataTaskWithURL(url: NSURL, completionHandler: (NSData?, NSURLResponse?, NSError?) -> Void = { (_, _, _) in }) -> NSURLSessionDataTask {
  ...
```

prevents compilation at the call sites without the `completionHandler` argument (e.g., `TurntableTests.swift:387:20: Ambiguous use of 'dataTaskWithURL(_:completionHandler:)'`).

This compiler error is completely reasonable and makes sense to me since `NSURLSession` defines 4 method signatures for ["Adding Data Tasks to a Session"](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/doc/uid/TP40013435-CH1-SW15). By providing a default value the signature now is ambiguous with the base class method signature.

There are 4 methods:

* [`dataTaskWithURL(_:)`](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/occ/instm/NSURLSession/dataTaskWithURL:)
* [`dataTaskWithURL(_:completionHandler:)`](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/occ/instm/NSURLSession/dataTaskWithURL:completionHandler:)
* [`dataTaskWithRequest(_:)`](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/occ/instm/NSURLSession/dataTaskWithRequest:)
* [`dataTaskWithRequest(_:completionHandler:)`](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/occ/instm/NSURLSession/dataTaskWithRequest:completionHandler:)

This means that it is not so much cluttering the API with extra methods as much as, as the `override` keyword denotes, overriding the implementation of the base class.